Collectors/DetailedCollector.py

```diff
@@ -374,15 +382,51 @@ def process(self, data, addr, port):
             pass
 
         elif header.code == b'g':
-            # The rest of the message is the gstream event
             self.logger.debug("Received gstream message")
+
+            userfromserver = self._users[sid].items()[0];
```

Comment:
I think you will need to check if the `sid` exists in the `_users` dict before trying to access it.  Also, this is not how you get the user information.  You get the user information like the following:

https://github.com/opensciencegrid/xrootd-monitoring-collector/blob/ab8a2af1b48bf3372bcaebcba8d66fd3b2822b80/Collectors/DetailedCollector.py#L65-L70